src/activity_handlers.cpp

```diff
@@ -3216,7 +3216,8 @@ void activity_handlers::find_mount_do_turn( player_activity *act, player *p )
         return;
     }
     npc &guy = dynamic_cast<npc &>( *p );
-    monster *mon = guy.chosen_mount.lock().get();
+    auto strong_monster = guy.chosen_mount.lock();
+    monster *mon = strong_monster.get();
```

Comment:
I believe you can just do
```cpp
std::shared_ptr<monster> mon = guy.chosen_mount.lock();
```
and remove `strong_monster` (pun intended?) altogether.

Reply:
Pun was intended yes :P